pennylane/templates/subroutines/commuting_evolution.py

```diff
@@ -124,25 +122,53 @@ def __init__(self, hamiltonian, time, frequencies=None, shifts=None, do_queue=Tr
             self.grad_recipe = (recipe,) + (None,) * len(hamiltonian.data)
             self.grad_method = "A"
 
-        self.hamiltonian = hamiltonian
-        self.num_params = len(hamiltonian.data) + 1

```

Comment:
No, I think it is now inferred from `len(self.parameters)` by the property of the same name

Reply:
oh, it's set in the parent class? I was confused reading the source code trying to work this out 😆

Reply:
It is not obvious here! This is one of the cases where we needed to overwrite the arbitrary `num_params` attribute, which is not necessary any more since `num_params` is an instance property.